File: quality_code_release_v1.0/dct1.m
function [X A] = dct1(x, k_max) 
%%% X is the first k_max coefficients of the DCT of x
%%% A is the DCT transformation matrix
N = size(x, 1);

if ~exist('k_max')
  k_max = N;
else
  k_max = min(k_max, N);
end
 
A = cos(pi/N*[0:k_max-1]'*([0:N-1]+.5));
X = A*x;

File: dct_svr/dct1.py
from math import *
from numpy import *

# useful link: https://numpy.org/doc/stable/user/numpy-for-matlab-users.html

def dct1(x, k_max=infty):

    N = x.shape[0]
    k_max = min(k_max, N)

    A = cos((pi/N*arange(0.,k_max)[:, newaxis])*(arange(0.,N-1)+5))
    # not too sure about this transpose... but it does make the error go away :D
    X = A.transpose()*x

    return A, X